### ISSUE TITLE
no-extra-parens ignores template literal's expressions

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#11906]
**Tell us about your environment**

* **ESLint Version:**  6.0.0 (same with 5.16.0)
* **Node Version:** 10.16.0
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  parserOptions: {
    ecmaVersion: 6,
  },
  rules: {
    "no-extra-parens": "error",
  }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
let foo = `${(a)}`
```

```bash
eslint index.js
```

**What did you expect to happen?**

1 error.

**What actually happened? Please include the actual, raw output from ESLint.**

Nothing, no errors.

**Are you willing to submit a pull request to fix this bug?**

Yes, I'll do it.

Inner expressions are fine (e.g. `((b * c) + d)` would be reported as one error), only the top level is missing.

Found this while working on [#11848](https://github.com/eslint/eslint/pull/11848), the `"TemplateLiteral"` handler is missing. It should check its `expressions`.

Is there a single case where these parens could be necessary?
